src/graphql/data/TopTokens.ts

```diff
@@ -135,16 +130,31 @@ interface UseTopTokensReturnValue {
   sparklines: SparklineMap
 }
 
+function replaceWrappedWithNative<T extends { address: string | null } | null>(chainId: number, token: T): T {
+  if (!token?.address) return token
+  const address = token.address.toLowerCase()
+  const nativeAddress = WRAPPED_NATIVE_CURRENCY[chainId]?.address.toLowerCase()
+  if (address !== nativeAddress) return token
+
+  const nativeToken = nativeOnChain(chainId)
+  return { ...token, ...nativeToken, address: 'NATIVE' }
+}
+
 export function useTopTokens(chain: Chain): UseTopTokensReturnValue {
+  const chainId = CHAIN_NAME_TO_CHAIN_ID[chain]
   const duration = toHistoryDuration(useAtomValue(filterTimeAtom))
 
   const environment = useRelayEnvironment()
   const [sparklines, setSparklines] = useState<SparklineMap>({})
-  useMemo(() => {
-    fetchQuery<TopTokensSparklineQuery>(environment, tokenSparklineQuery, {
-      duration,
-      chain,
-    }).subscribe({
+  const query = useMemo(
+    () =>
+      fetchQuery<TopTokensSparklineQuery>(environment, tokenSparklineQuery, { duration, chain }).map((data) => ({
+        topTokens: data.topTokens?.map((token) => replaceWrappedWithNative(chainId, token)),
+      })),
+    [chain, chainId, duration, environment]
+  )
+  useEffect(() => {
+    const subscription = query.subscribe({
```

Comment:
this memo -> useEffect link feels off

They're tightly coupled, but the code doesn't really make that clear until you take a sec to read it. Would putting the query construction currently in useMemo in the useEffect below have any negative effect? All else equal I'd prefer the query stays with its usage.

Reply:
I wrote the original useMemo here. Since useLazyLoadQuery suspends, the first render triggering a useEffect wouldn't run until that query finishes. For perf, I put it in a memo so the two queries run in parallel. I agree that it looks wonky tho

Reply:
I agree - this is an artifact of refactoring.